internal/wire/frame_parser.go

```diff
@@ -11,6 +11,8 @@ import (
 )
 
 type frameParser struct {
+	r *bytes.Reader // cached bytes.Reader, so we don't have to repeatedly allocate them
```

Comment:
Nit: You can probably also avoid the ptr indirection here.

Reply:
Done.